discover/portal/articles/03-collaboration/01-managing-documents-and-media/08-adapting-your-media.markdown

```diff
@@ -1,308 +1,165 @@
-variant. The UUID must be a unique identifier for that particular
-variant; there must be no two variants with the same UUID. Finally, the
-set of attribute/value pairs must be consistent with the attributes
-defined in the subtypes of AdaptiveMediaAttribute; for example, in the
-case of adaptive images the valid attributes are “widht” and “height”.
-
-This is an example of a valid configuration that will create images with
-a maximum height of 100px and a maximum width of 200px:
-
-lowresolution:12345:height=100;width=200
-
-Adaptive Media Image Attributes {#h.lkjxewowpv4h .c1 .c12}
--------------------------------
-
-The information of the adaptive images can be obtained using Adaptive
-Media Attributes. There are a few generic Adaptive Media attributes that
-are common to every “media” and there are a few specific attribute for
-images:
-
-com.liferay.adaptive.media.AdaptiveMediaAttribute contains the
-attributes common to every media. They are:
-
--   Content Length
--   Content Type
--   File Name
+-   Very High Resolution  max-width: 3000px; max-height: 2400px;
+-   High Resolution  max-width: 2400px; max-height: 1800px;
+-   Medium Resolution  max-width: 1200px; max-height: 800px;
+-   Low Resolution  max-width: 400px; max-height: 200px;
 
-com.liferay.adaptive.media.image.processor.ImageAdaptiveMediaAttribute
-contains the attributes specific to images:
+With this configuration, for every image that is added to a Documents and Media 
+repository that supports Adaptive Media (such as Documents and Media repository, 
+Blogs repository, etc), four new image variants will be created with the 
+max-width and max-height specified. Note that generated images are scaled to 
+best fit the desired resolutions while still maintaining their original aspect 
+ratio, so the generated images may not match the exact width/height specified.
 
--   Height
--   Width
+The Adaptive Media's API determines which image variant is best to use for the
+given device's specifications. For example, if a smart watch is viewing the
+image, given the configuration above, the Low Resolution image is displayed. If
+the image is viewed on a desktop with a large screen instead, the Very High 
+Resolution image is displayed.
 
-This attributes can be used to obtain the value of a particular media as
-follows:
+You can specify the image variant to use for a device in a few different ways:
 
-adaptiveMedia.getAttributeValue(
+-   Choose the image whose width is smaller than the screen width.
+-   Choose the image whose height is smaller than the screen height.
+-   Choose the image whose width and height are smaller than the
+    screen width and height.
+-   Choose the image whose width and height are more similar to the
+    screen size.
+-   Choose the image whose width and height are slightly smaller than
+    the screen size
 
-  ImageAdaptiveMediaAttribute.IMAGE\_HEIGHT)
+As you can see, there are many ways to choose the best image for a device. The 
```

Comment:
Hey @mwilliams2014 the consumer can specify what the query provides. Take a look at the tests here https://github.com/liferay/com-liferay-adaptive-media/blob/master/adaptive-media-image-impl/src/test/java/com/liferay/adaptive/media/image/internal/finder/ImageAdaptiveMediaQueryBuilderImplTest.java#L72

You can see that the query is implemented using a Builder pattern that "tells" the developer the different options available.